File: engine/core/ids.py
"""ID helpers for battle entities and events."""

from __future__ import annotations

import re

_ID_PATTERN = re.compile(r"^[a-z][a-z0-9_-]{1,63}$")


def is_valid_id(value: str) -> bool:
    return bool(_ID_PATTERN.match(value))


def event_id(sequence: int) -> str:
    return f"ev_{sequence:06d}"

File: old/src/engine/ids.ts
/**
 * ID helpers for battle entities and events.
 * Mirrors engine/core/ids.py
 */

const ID_PATTERN = /^[a-z][a-z0-9_-]{1,63}$/;

export function isValidId(value: string): boolean {
  return ID_PATTERN.test(value);
}

export function eventId(sequence: number): string {
  return `ev_${String(sequence).padStart(6, "0")}`;
}
